feature_engine/encoding/base_encoder.py

```diff
@@ -124,7 +138,8 @@ def _check_or_select_variables(self, X: pd.DataFrame):
             self.variables_ = _find_all_variables(X, self.variables)
 
         # check if dataset contains na
-        _check_contains_na(X, self.variables_)
+        if not self.allow_missing:
```

Comment:
I created the _fit method and removed the check_na from this method, so you can use it in MatchVariables.

In short, this logic that you are using here, you need to move it to MatchVariables.